tests/test_model_consistency.py

```diff
@@ -230,3 +269,70 @@ def get_connected(diagram, handle):
 
 def ordered(elements):
     return sorted(elements, key=lambda e: e.id)  # type: ignore[no-any-return]
+
+
+@singledispatch
+def check_relation(relation: object, head, tail):
+    assert False, f"No comparison function for {relation}"
+
+
+@check_relation.register
+def _(relation: diagramitems.DependencyItem, head, tail):
```

Comment:
I've done that basically since I was too lazy to come up with a proper name. This is just a singledispatch function. Has nothing to do with Hypothesis.